diff-cam-engine.js

```diff
@@ -105,9 +107,12 @@ const DiffCamEngine = (function () {
       };
     }
 
+    // check if a special deviceId was passed as an option
+    const userMediaOptions = deviceId ? { video: { deviceId: { exact: deviceId } } } : { video: true };
```

Comment:
Sorry for the delay, but I was able to test this today on a spare RPI 3 with one webcam. I got the device ID and configured it and kept getting an 'OverconstrainedError' displayed on the MM screen. My background is in pure java, and not JS so I'm not sure why its throwing that error. I was however, able to get it to work by removing the "exact:" constraint. The documentation I found on the link below shows you can leave "exact:" out and use the "preferred" deviceId. In doing this...it works. Tomorrow I might be able to get access to my MM pie that has 2 cameras and test it out there.

https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia

Reply:
I can confirm that 2 camera devices will work with the changes specifying the 'preferred' device. Not sure why the 'exact' constraint doesn't work.

Reply:
Without the exact contrain the userMedia would use any cam as a fallback if it doesnt find the one specified.

Did you retrieve the ID from the electron app of the MagicMirror? The ids change if you retrieve them from another browser.

Reply:
I've been digging into that today, and did not know electron had different IDs than Chromium browser. I've got some code I ran in a Chromium browser IDE that got me IDs. How can I get the IDs through the MM electron app? I've tried putting the code into your code and doing a console.log with the IDs...but nothing ever gets written to the log. In fact, the only place that seems to log is he node_helper.js.

Reply:
Well, finally figured out how to get the logs I was creating. Needed to add 'export ELECTRON_ENABLE_LOGGING=true' to my mm.sh. Then cat the pm2 error logs, and that gave me the DeviceID. Using that ID no longer threw the constraint error and it uses the correct Camera.